### PR TITLE
IBC Client as dependency checks

### PR DESCRIPTION
**Pull request by Buckram123**
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-sdk/src/base/endpoints/modules_ibc.rs [hunk 1]
```diff
@@ -16,6 +17,19 @@ pub trait ModuleIbcEndpoint: Handler {
         info: MessageInfo,
         msg: ModuleIbcMsg,
     ) -> Result<Response, Self::Error> {
+        // Make sure module have ibc_client as dependency
+        if !self
```
Comment (CyberHoward) [inline framework/packages/abstract-sdk/src/base/endpoints/modules_ibc.rs:21]: When a module receives an IBC message it isn't required to have a dependency on the IBC client (the message is sent from the host)

**Review by CyberHoward — CHANGES_REQUESTED** [review]
Sweet! I'd like us to also add the assertion in the IBC APIs, like here: https://docs.rs/abstract-sdk/latest/src/abstract_sdk/apis/app.rs.html#78

File: framework/packages/abstract-sdk/src/apis/ibc.rs [hunk 2]
```diff
@@ -87,11 +87,11 @@ pub struct IbcClient<'a, T: IbcInterface> {
 impl<'a, T: IbcInterface> IbcClient<'a, T> {
     /// Get address of this module
     pub fn module_address(&self) -> AbstractSdkResult<Addr> {
```
Comment (CyberHoward) [inline framework/packages/abstract-sdk/src/apis/ibc.rs:89]: Could we rename this to `address` instead of `module_address`?

**Review by CyberHoward — APPROVED** [review]

File: framework/packages/abstract-sdk/src/apis/ibc.rs (hunk 2 above)
Comment (Buckram123) [reply framework/packages/abstract-sdk/src/apis/ibc.rs:89]: We use `module_address` in `Modules` and `CronCat` API's as well! Would you prefer to use `address` there too?

**Pull request merged**